pkg/controller/acmpca/certificateauthority/controller.go

```diff
@@ -0,0 +1,278 @@
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package certificateauthority
+
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	awsacmpca "github.com/aws/aws-sdk-go-v2/service/acmpca"
+	"github.com/google/go-cmp/cmp"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	runtimev1alpha1 "github.com/crossplane/crossplane-runtime/apis/core/v1alpha1"
+	"github.com/crossplane/crossplane-runtime/pkg/event"
+	"github.com/crossplane/crossplane-runtime/pkg/logging"
+	"github.com/crossplane/crossplane-runtime/pkg/meta"
+	"github.com/crossplane/crossplane-runtime/pkg/reconciler/managed"
+	"github.com/crossplane/crossplane-runtime/pkg/resource"
+
+	v1alpha1 "github.com/crossplane/provider-aws/apis/acmpca/v1alpha1"
+	acmpca "github.com/crossplane/provider-aws/pkg/clients/acmpca"
+	"github.com/crossplane/provider-aws/pkg/controller/utils"
+)
+
+const (
+	errPendingStatus    = "The managed resource in pending status, please open the ACM Private CA console https://console.aws.amazon.com/acm-pca/home install CA certificate "
+	errUnexpectedObject = "The managed resource is not an ACMPCA resource"
+	errClient           = "cannot create a new ACMPCA client"
+	errGet              = "failed to get ACMPCA with name"
+	errCreate           = "failed to create the ACMPCA resource"
+	errDelete           = "failed to delete the ACMPCA resource"
+	errEmpty            = "empty ACMPCA received from ACMPCA API"
+
+	errKubeUpdateFailed    = "cannot late initialize ACMPCA"
+	errUpToDateFailed      = "cannot check whether object is up-to-date"
+	errPersistExternalName = "failed to persist Certificate ARN"
+
+	errAddTagsFailed        = "cannot add tags to ACMPCA"
+	errListTagsFailed       = "failed to list tags for ACMPCA"
+	errRemoveTagsFailed     = "failed to remove tags for ACMPCA"
+	errCertificateAuthority = "failed to update the ACMPCA resource"
+)
+
+// SetupCertificateAuthority adds a controller that reconciles ACMPCA.
+func SetupCertificateAuthority(mgr ctrl.Manager, l logging.Logger) error {
+	name := managed.ControllerName(v1alpha1.CertificateAuthorityGroupKind)
+
+	return ctrl.NewControllerManagedBy(mgr).
+		Named(name).
+		For(&v1alpha1.CertificateAuthority{}).
+		Complete(managed.NewReconciler(mgr,
+			resource.ManagedKind(v1alpha1.CertificateAuthorityGroupVersionKind),
+			managed.WithExternalConnecter(&connector{client: mgr.GetClient(), newClientFn: acmpca.NewClient, awsConfigFn: utils.RetrieveAwsConfigFromProvider}),
+			managed.WithConnectionPublishers(),
+			managed.WithInitializers(),
```

Comment:
```suggestion
                      // TODO: implement tag initializer
			managed.WithInitializers(),
```

I can follow up and add this, but would add a TODO here please.

Reply:
Added TODO comment